### PR TITLE
feat(zharta): deposits

### PR DESCRIPTION
**Pull request by DioPires**
## Description

Adds the Lending Pool deposits in Zharta

## Checklist

- [x] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)
- [x] (optional) As a contributor, my Ethereum address/ENS is: pirex.eth
- [x] (optional) As a contributor, my Twitter handle is: 0xPirex

### CODE REVIEW

**Comment by wpoulin** [comment]
Hey

Good job ! 
I'm guessing what's next is you would like loans originated on Zharta to show up Zapper right ? If so, I'll have to implement it myself in one of our private repo because our NFT service isn't exposed externally. That would also mean I will have to implement (aka copy pasta) this PR in our private repo as well. 

In the meantime, in order to facilitate integration, could you verify the contracts listed in your docs 🙏 
https://zharta.gitbook.io/zharta-welcome-kit/protocol/contract-addresses

**Comment by DioPires** [comment]
@wpoulin thanks for checking the PR 🙌

I'll verify the contracts. Apart from that, do you need any support from our side?

**Comment by DioPires** [comment]
@wpoulin the first 4 contracts [here](https://zharta.gitbook.io/zharta-welcome-kit/protocol/contract-addresses) are already verified, which are from where the events regarding deposits/withdrawals and loans come from. Do you need more for the integration?

**Comment by wpoulin** [comment]
I've been looking at the 4 verified contracts and I was wondering how can I retrieve the loanIds for a given address. I've also had a look at the contracts which weren't verified on Github and didn't find any contract with `balanceOf` and `tokenOfOwnerByIndex` which would allow retrieving loanIds for a given wallet

**Comment by DioPires** [comment]
@wpoulin I'm afraid there are no such methods in the loans contracts. That data is indexed off-chain.

If you can't index it off-chain, I guess the only way to know which loans have already been started by a given wallet is to iterate over possible loan ids (from `0` onwards) until an empty object is returned. Once that happens, it means there are no more loans for that wallet.

**Comment by wpoulin** [comment]
ahh okok 
I'll merge this PR

File: src/apps/zharta/ethereum/zharta.lending-pool-core.contract-position-fetcher.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+} from '~position/template/contract-position.template.types';
+
+import { ZhartaContractFactory, ZhartaLendingPoolCore } from '../contracts';
+
+interface ZhartaLendingPoolCoreContractPositionDefinition extends DefaultContractPositionDefinition {
+  type: 'LENDING_POOL_CORE';
+}
+
+@PositionTemplate()
+export class EthereumZhartaLendingPoolCoreContractPositionFetcher extends ContractPositionTemplatePositionFetcher<Contract> {
```
Comment (wpoulin) [inline src/apps/zharta/ethereum/zharta.lending-pool-core.contract-position-fetcher.ts:27]: You want to extend the `Contract` you're calling, which in this case it `ZhartaLendingPoolCore`, same applies in the getContract, etc

File: src/apps/zharta/ethereum/zharta.lending-pool-core.contract-position-fetcher.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    @Inject(ZhartaContractFactory) protected readonly contractFactory: ZhartaContractFactory,
+  ) {
+    super(appToolkit);
+  }
+
+  getContract(_address: string): Contract {
+    return this.contractFactory.zhartaLendingPoolCore({ address: _address, network: this.network });
+  }
+
+  async getDefinitions() {
```
Comment (wpoulin) [inline src/apps/zharta/ethereum/zharta.lending-pool-core.contract-position-fetcher.ts:42]: You can add the return type to the functions signature, so you won't need to add types to the regularLendingPoolCoreDefinitions's declaration

**Review by wpoulin — APPROVED** [review]
Well done !

**Pull request merged**